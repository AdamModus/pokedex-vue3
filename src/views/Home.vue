<template>
  <div class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-4 xl:grid-cols-6">
    <PokemonCard
      v-for="pokemon in pokemonStore.allPokemons"
      :key="pokemon.id"
      :id="pokemon.id"
      :name="pokemon.name"
      :url="pokemon.url"
      :imgUrl="pokemon.imageUrl"
    >
    </PokemonCard>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue';
import { usePokemonStore } from '@/store';
import PokemonCard from '@/components/PokemonCard.vue'; // @ is an alias to /src

export default defineComponent({
  name: 'Home',
  components: {
    PokemonCard,
  },

  setup() {
    const pokemonStore = usePokemonStore();

    return { pokemonStore };
  },

  mounted() {
    this.pokemonStore.fetchAllPokemons();
  },
});
</script>
