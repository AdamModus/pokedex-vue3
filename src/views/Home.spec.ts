import { createTestingPinia } from '@pinia/testing';
import { shallowMount, VueWrapper } from '@vue/test-utils';
import Home from './Home.vue';

describe('Home.vue', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    const mounOptions = {
      global: {
        plugins: [createTestingPinia()],
      },
    };
    wrapper = shallowMount(Home, mounOptions);
  });

  it('should the Home View', () => {
    // Assert
    expect(wrapper.element).toMatchSnapshot();
  });
});
