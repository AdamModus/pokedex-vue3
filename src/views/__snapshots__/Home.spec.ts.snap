// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue should the Home View 1`] = `
<div
  class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-4 xl:grid-cols-6"
>
  
  
</div>
`;
